src/v/redpanda/monitor_unsafe_log_flag.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/types.h"
+
+class monitor_unsafe_log_flag {
+public:
+    static constexpr ss::shard_id backend_shard = 0;
+    // Flag introduced in version v23.2.1 (cluster version 10)
+    static constexpr cluster::cluster_version flag_introduction_version
+      = cluster::cluster_version{10};
+    monitor_unsafe_log_flag(
+      ss::abort_source& shard0_as, cluster::cluster_version original_version);
+
+    ss::future<> start();
+    ss::future<> stop();
+
+private:
+    class handle_unsafe_log_flag_change;
+
+    ss::future<> maybe_log_flag_nag();
+
+    ss::abort_source& _shard0_as;
+    cluster::cluster_version _original_version;
+    ss::gate _gate;
+    ss::sharded<handle_unsafe_log_flag_change> _handle_unsafe_log_flag_change;
```

Comment:
what is `ss::sharded` needed for? it seems like all of the meaningful work is happening on core 0 and every other core only needs to touch the thread local flag?